src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -240,8 +179,18 @@ private async Task ProcessAsynchronousWork()
 
                 // new work might be added before we check again as a result of waiting for all
                 // the child components to finish executing SetParametersAsync
-                await pendingWork;
-            };
+
+                try
+                {
+                    await pendingWork;
+                }
+                catch when (!pendingWork.IsCanceled)
+                {
+                    // await will unwrap an AggregateException and throw exactly one inner exception.
+                    // Using pendingWork.Exception gives us access to all of the exception
+                    HandleException(pendingWork.Exception);
+                }
```

Comment:
This try-catch block probably needs to be done before the Task.WhenAll so that we report on the individual exception.